File: C++_version/Process.cpp
#include "Process.hpp"

int ID = 1;

Process::Process(int arrival_time,
                 int burst_time,
                 int priority) : id(ID), arrival_time(arrival_time), burst_time(burst_time), priority(priority)
{
    ID++;
    execution_time = 0;
    waiting_time = 0;
    turnaround_time = 0;
};
int Process::getID() const
{
    return id;
}

int Process::getPriority() const
{
    return priority;
}

int Process::getArrival_time() const
{
    return arrival_time;
}

int Process::getBurst_time() const
{
    return burst_time;
}

int Process::getExecution_time() const
{
    return execution_time;
}

int Process::getwaiting_time() const
{
    return waiting_time;
}

int Process::getturnaround_time() const
{
    return turnaround_time;
}

void Process::setID(int id)
{
    this->id = id;
}

void Process::setPriority(int priority)
{
    this->priority = priority;
}

void Process::setArrival_time(int arrival_time)
{
    this->arrival_time = arrival_time;
}

void Process::setBurst_time(int burst_time)
{
    this->burst_time = burst_time;
}

void Process::setExecution_time(int execution_time)
{
    this->execution_time = execution_time;
}

void Process::setwaiting_time(int waiting_time)
{
    this->waiting_time = waiting_time;
}
void Process::setturnaround_time(int turnaround_time)
{
    this->turnaround_time = turnaround_time;
}

File: java_version/Process.java
public class Process {
    private static int ID_COUNTER = 1;
    private int id;
    private int priority;
    private int arrivalTime;
    private int burstTime;
    private int executionTime;
    private int waitingTime;
    private int turnaroundTime;

    public Process(int arrivalTime, int burstTime, int priority) {
        this.id = ID_COUNTER++;
        this.arrivalTime = arrivalTime;
        this.burstTime = burstTime;
        this.priority = priority;
        this.executionTime = 0;
        this.waitingTime = 0;
        this.turnaroundTime = 0;
    }

    public int getId() {
        return id;
    }

    public int getPriority() {
        return priority;
    }

    public int getArrivalTime() {
        return arrivalTime;
    }

    public int getBurstTime() {
        return burstTime;
    }

    public int getExecutionTime() {
        return executionTime;
    }

    public int getWaitingTime() {
        return waitingTime;
    }

    public int getTurnaroundTime() {
        return turnaroundTime;
    }

    public void setPriority(int priority) {
        this.priority = priority;
    }

    public void setExecutionTime(int executionTime) {
        this.executionTime = executionTime;
    }

    public void setWaitingTime(int waitingTime) {
        this.waitingTime = waitingTime;
    }

    public void setTurnaroundTime(int turnaroundTime) {
        this.turnaroundTime = turnaroundTime;
    }
}
